addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.js

```diff
@@ -0,0 +1,11 @@
+/* @odoo-module */
+
+import { DiscussSidebarCategories } from "@mail/discuss/core/web/discuss_sidebar_categories";
+import { patch } from "@web/core/utils/patch";
+
+patch(DiscussSidebarCategories.prototype, {
+    filteredThreads(category) {
+        const threads = super.filteredThreads(category);
+        return threads.filter((thread) => !thread.readonly);
+    },
+});
```

Comment:
filtering of channels in sidebar should normally be done with existence of channel.member for current user + is_pinned

Reply:
Could you please elaborate on this ? My code is not meant to alter the existing logic you mentionned. The only thing it does is adding another condition on top of it.

Reply:
@seb-odoo I investigated this a little bit more. As a reminder, we talked about it orally yesterday and you explained that the condition should include the fact that the user has to be a member of the channel.

In mail/, this condition is not necessary because it is stille required to be a member to fetch channel information.
```python
    @http.route("/discuss/channel/info", methods=["POST"], type="json", auth="public")
    @add_guest_to_context
    def discuss_channel_info(self, channel_id):
        member_sudo = request.env["discuss.channel.member"]._get_as_sudo_from_context(channel_id=int(channel_id))
        return member_sudo.channel_id._channel_info()
```

But in im_livechat/ I modified this route slightly and it is possible for livechat administrators to fetch it.
```python
    @route()
    def discuss_channel_info(self, channel_id):
        info = super().discuss_channel_info(channel_id)
        if not info and request.env.user.has_group("im_livechat.im_livechat_group_manager"):
            channel = request.env["discuss.channel"]._get_from_context_or_raise((channel_id))
            info = channel._channel_info({"readonly": True})
        return info
```

However, I also modified the filter because it is specified in the task that we want to browse easily between readonly channels when Discuss was opened from the history.
```javascript
patch(DiscussSidebarCategories.prototype, {
    filteredThreads(category) {
        const threads = super.filteredThreads(category);
        return threads.filter((thread) => this.store.discuss.livechatHistory || !thread.readonly);
    },
});
```

I know you wanted me to get rid of the override of ```filteredThreads()``` but I don't think it will be possible even if I modify the base implementation of this method to double check that the user is a member.